File: python/euler/001.py
def divisible_by(list_, factors):
    filtered_list = []
    for element in list_:
        for factor in factors:
            if element % factor == 0:
                filtered_list.append(element)
                break
    return filtered_list

if __name__ == "__main__":
    sum_ = sum(divisible_by(range(0, 1000), [3,5]))
    print(f"sum = {sum_}")

File: julia/euler/001.jl
function divisible_by(list::StepRange{Int,Int}, factors::Array{Int,1})::Array{Int, 1}
  filtered_list::Array{Int} = []
  for element in list
    for factor in factors
      if element % factor == 0 
        append!(filtered_list, element)
        break
      end
    end
  end
  return filtered_list
end

println("sum(divisible_by(range(0,1000,step=1), [3,5]) = $(sum(divisible_by(range(0,9,step=1), [3,5])))")
